### ISSUE TITLE
No context about filegroups

### ISSUE THREAD

**Issue by DaveBoltman** [MicrosoftDocs/sql-docs#9351]
When I use [Example A](https://learn.microsoft.com/en-us/sql/relational-databases/databases/create-a-database-snapshot-transact-sql?view=sql-server-ver16#a-create-a-snapshot-on-the-adventureworks-database) on the page on my own database (not the AdventureWorks database), I get an error "Msg 5127, Level 16, State 1, Line 1 - All files must be specified for database snapshot creation." My database has custom filegroups, so clearly there's something this page is not telling me! The word "filegroup" does not appear on the page in any way that helps me resolve the problem. My guess is that the following information on the page is plain wrong, since AFAIK SQL Server databases do not have logical names, and that "logical_file_name" is not what the page says, but something to do with filegroups.

<html>
<body>

Argument | Description
-- | --
logical_file_name | The logical name of the source database used in SQL Server when referencing the file.

</body>
</html>

[Another page](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-database-transact-sql?view=sql-server-ver16&tabs=sqlpool#f-create-a-database-snapshot) is more helpful regarding creating snapshots, by saying "In conformance with the syntax, every file in the source database is specified, and filegroups are not specified". How about mentioning that on this page as well, to make it clear and avoid hiding that critical information in the noise?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: f75a530e-446a-a521-c186-488cbed9b463
* Version Independent ID: 417b56a6-e21c-c7d1-8fe9-04caf78fd3a4
* Content: [Create a database snapshot (Transact-SQL) - SQL Server](https://learn.microsoft.com/en-us/sql/relational-databases/databases/create-a-database-snapshot-transact-sql?view=sql-server-ver16)
* Content Source: [docs/relational-databases/databases/create-a-database-snapshot-transact-sql.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/databases/create-a-database-snapshot-transact-sql.md)
* Service: **sql**
* Sub-service: **configuration**
* GitHub Login: @WilliamDAssafMSFT
* Microsoft Alias: **wiassaf**

**Comment by WilliamAntonRohm** [comment]
@DaveBoltman -- Dave, thank you for your thorough feedback.  Please consider these resources:
- [SQL Server on Microsoft Q&A](https://docs.microsoft.com/answers/products/sql-server)
- [DBA Stack Exchange](https://dba.stackexchange.com/questions/tagged/sql-server)
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sql-server)

@WilliamDAssafMSFT -- William, please look into this issue.

**Comment by rwestMSFT** [comment]
#reassign:rwestMSFT

Thanks @DaveBoltman, we'll look into clarifying the language.

**Comment by rwestMSFT** [comment]
@DaveBoltman After reviewing your comments and the article in question, I will make the following change to the main **Examples** heading (My emphasis in bold).

> The `.ss` extension used in these examples is for convenience, and isn't required. **In databases containing multiple files, all files must be specified, in conformance with the syntax. Filegroups aren't specified.**

Regarding your feedback, I'd like to point out that Example B covers the scenario you raised.

> AFAIK SQL Server databases do not have logical names, and that "logical_file_name" is not what the page says, but something to do with filegroups.

Every file in a SQL Server database has a logical name, irrespective of its filegroup and physical path. Filegroups themselves do not have file names to speak of, since they are a not much more than a metadata value. To understand this better, let's take the the [`WideWorldImporters`](https://github.com/Microsoft/sql-server-samples/releases/tag/wide-world-importers-v1.0) sample database as an example.

|Logical name|File type|Filegroup|
|---|---|---|
|WWI_Primary|ROWS data|PRIMARY|
|WWI_UserData|ROWS data|USERDATA|
|WWI_InMemory_Data1|FILESTREAM data|WWI_InMemory_Data|
|WWI_Log|LOG|N/A|

If I add a new file to the `USERDATA` filegroup, the logical name I provide has no bearing on the filegroup itself. If I call the new file `WWI_PrimaryData_3` but assign it to the `USERDATA` filegroup, I will see the following new file in the Files output in SSMS:

|Logical name|File type|Filegroup|
|---|---|---|
|WWI_PrimaryData_3|ROWS data|USERDATA|

When I want to create a snapshot, it's this list of logical file names I need to use, and that is expressed clearly in Example B.

To summarize, filegroups are not a function of logical names in any respect. The documentation is correct as written, but I think adding the clarifying language in the **Examples** section will help.

**Comment by DaveBoltman** [comment]
Thank you, @rwestMSFT, that's certainly clearer. But, I think one of the items I mentioned still causes confusion:

<html>
<body>


Argument | Description
-- | --
logical_file_name | The logical name of the **source database** used in SQL Server when referencing the file.



</body>
</html>

Of course you're right about every file having a logical name, but the docs say "logical name of the _source database_ used .....", not the "source database file used ....." To me it's saying that the database itself (not it's files) has a logical name. Perhaps one of the following would make it clear?

- "The logical name of the source database _file_ used in SQL Server when referencing the file"
- "The logical name ~~of~~ the source database _uses_ in SQL Server when referencing the file"

Regards,
Dave

**Comment by rwestMSFT** [comment]
@DaveBoltman Great feedback, thank you. I'll correct that description.
